x/wasm-storage/keeper/filter.go

```diff
@@ -0,0 +1,147 @@
+package keeper
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/tidwall/gjson"
+)
+
+const (
+	None         = iota
+	Mode         = 1
+	StdDeviation = 2
+)
+
+const (
+	INT uint8 = iota + 1
+	INT8
+	INT16
+	INT32
+	INT64
+	UINT
+	UINT8
+	UINT16
+	UINT32
+	UINT64
+	FLOAT32
+	FLOAT64
+)
+
+type (
+	filterProp struct {
+		Algo       uint
+		JSONPath   string `rlp:"optional"`
+		MaxSigma   uint64 `rlp:"optional"`
+		NumberType uint8  `rlp:"optional"`
+	}
+)
+
+func FilterMode(jsonPath string, numberType uint8, exitCodes []uint8, reveals [][]byte) ([]bool, bool) {
+	var (
+		outliers        []bool
+		consensus       bool
+		nonZeroExitCode int
+	)
+	results := make([]gjson.Result, 0, len(reveals))
+
+	for i, r := range reveals {
+		results = append(results, gjson.GetBytes(r, jsonPath))
+		if exitCodes[i] != 0 {
+			nonZeroExitCode++
+		}
+	}
+
+	switch numberType {
+	case INT, INT8, INT16, INT32, INT64:
+		vals := make([]int64, 0, len(reveals))
+		for _, res := range results {
+			vals = append(vals, res.Int())
+		}
+		outliers, consensus = calculate(vals)
+	case UINT, UINT8, UINT16, UINT32, UINT64:
+		vals := make([]uint64, 0, len(reveals))
+		for _, res := range results {
+			vals = append(vals, res.Uint())
+		}
+		outliers, consensus = calculate(vals)
+	case FLOAT32, FLOAT64:
+		vals := make([]float64, 0, len(reveals))
+		for _, res := range results {
+			vals = append(vals, res.Float())
+		}
+		outliers, consensus = calculate(vals)
+	default:
+		vals := make([]string, 0, len(reveals))
+		for _, res := range results {
+			vals = append(vals, res.String())
+		}
+		outliers, consensus = calculate(vals)
+	}
```

Comment:
We can't do in general `res.String()` instead of handling each case? Don't go have some interface where you can always convert everything to String?

Reply:
here: [92b66cf](https://github.com/sedaprotocol/seda-chain/pull/280/commits/92b66cfd5aacfb22659422df22d0eb81b458e261)